src/tasks/detection_task.py

```diff
@@ -0,0 +1,164 @@
+from typing import DefaultDict
+import torch
+from pydantic import BaseModel
+
+from src.constructor import create_backbone, create_scheduler, create_optimizer
+from src.constructor.config_structure import TrainConfigParams
+from src.registry import TASKS, DETECTION_NECKS, \
+    DETECTION_HEADS, DETECTION_HAT
+from .base_task import BaseTask
+import torch.nn as nn
+from src.models.backbones.utils.hub import download_cached_file
+
+class DetectionTaskParams(BaseModel):
+    checkpoint: str = None
+    input_size: list  # [num_channels, height, width]
+    freeze_backbone: bool = False
+    backbone_name: str
+    backbone_params: dict = {}
+    neck_name: str
+    neck_params: dict = {}
+    head_name: str
+    head_params: dict = {}
+    hat_name: str
+    hat_params: dict = {}
+
+
+@TASKS.register_class
+class DetectionTask(BaseTask, nn.Module):
+    config_parser = DetectionTaskParams
+
+    def __init__(self, hparams: TrainConfigParams):
+        super(DetectionTask, self).__init__(hparams)
+        # super().__init__(hparams)
+        self.backbone = create_backbone(model_name=self.params.backbone_name,
+                                        **self.params.backbone_params)
+        
+        neck_class = DETECTION_NECKS.get(self.params.neck_name)
+        self.neck = neck_class(
+                            in_channels=[128, 256, 512],
+                            out_channels=128,
+                            num_csp_blocks=1
+        )
+        
+        head_class = DETECTION_HEADS.get(self.params.head_name)
+        self.head = head_class(
+                            num_classes=self.params.head_params['num_classes'], 
+                            in_channels=128, 
+                            feat_channels=128
+        )
+
+        infer_class = DETECTION_HAT.get(self.params.hat_name)
```

Comment:
`hat_class`

Reply:
its already named backbone _name, neck_name and hat_name, if change hat_name to hat class we need to change backbone_name to backbone_class